core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -213,6 +213,28 @@ class KryoSerializer(conf: SparkConf)
     // We can't load those class directly in order to avoid unnecessary jar dependencies.
     // We load them safely, ignore it if the class not found.
     Seq(
+      "org.apache.spark.graphx.Edge",
+      "org.apache.spark.graphx.Edge$mcB$sp",
```

Comment:
These are synthetic classes and their name may change. Are they needed?

Reply:
Yes, they are needed if we want to register `Edge`, since type specialization is used in [https://github.com/apache/spark/blob/master/graphx/src/main/scala/org/apache/spark/graphx/Edge.scala#L32](url)

I had test this, if we do not reigster `org.apache.spark.graphx.Edge$mcB$sp`, Edge[Boolean] will not be handled by kryo.